### PR TITLE
intitial commit

### PR DESCRIPTION
**Pull request by srinizodiac**

### CODE REVIEW

File: app/controllers/claiimInviteCode.js [hunk 1]
```diff
@@ -1,18 +1,157 @@
...
+  }
+
+  catch(error) {
+    callback(null,error)
+  }
+
+};
+
+/**
+ * findDeviceIdInLegacy
```
Comment (chronicalley) [inline app/controllers/claiimInviteCode.js]: The comment here does not match the method name.

File: app/controllers/claiimInviteCode.js [hunk 2]
```diff
@@ -1,18 +1,157 @@
 'use strict';
```
Comment (chronicalley) [inline app/controllers/claiimInviteCode.js]: I thought the directive from @dpansez was to use Typescript for all new code?

File: app/controllers/claiimInviteCode.js [hunk 3]
```diff
@@ -1,18 +1,157 @@
...
+ * updateDynamodbInviteCodeTable
+ * ----------------------------
+ * update the invite_code dynamoDb table
+ *
+ *  @param {String} deviceId the device id
+ * @param {Object} rubyRes object 
+ * @param {integer} rubyUserId rubyUserId
+ * @return null
+ */ 
+function updateDynamodbInviteCodeTable(deviceId,rubyRes,rubyUserId) {
```
Comment (chronicalley) [inline app/controllers/claiimInviteCode.js]: I'd recommend moving all code related to creating and updating invite codes to a standalone class (repository or service class).

File: app/controllers/claiimInviteCode.js [hunk 4]
```diff
@@ -1,18 +1,157 @@
...
+     }
+     
+     if(dynamoData.statuscode == 200) {
+       var updateDynamo = await updateDynamodbInviteCodeTable(deviceId,rubyRes,rubyUserId);
+     }
+
+     callback(null, updateDynamo);
+  }
+
+  catch(error) {
```
Comment (chronicalley) [inline app/controllers/claiimInviteCode.js]: We should move away from using callbacks in handlers and instead just directly return the status code. Example: https://github.com/Zodiac-Engineering/platform-websockets-api/blob/develop/src/routes/connect.js

File: app/controllers/inviteCodeGenerate.js [hunk 5]
```diff
@@ -2,19 +2,83 @@
...
+  return new Promise (function (resolve, reject) {
+    let result = {};
+    result.funct = 'createDynamoDbData';
+    try {
+        let params = {};
+        params.TableName = process.env.INVITECODE_TABLE;
+        params.Item = { 
+            deviceId: deviceId,
+            invite_code: inviteCode,
+            claimedBy: "",
```
Comment (chronicalley) [inline app/controllers/inviteCodeGenerate.js]: Can claimedBy and claimedAt just be null to start? Why empty string?

File: config.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+'use strict'
```
Comment (chronicalley) [inline config.js]: We seem to have two different ways of managing constants in serverless projects. A config.js and then actually configuration files (like this: https://github.com/Zodiac-Engineering/platform-onica-device-control-services/blob/master/tcx/env/dev-us-west-2.yml).My preference would be YAML as it's easier to read. CC @Narayan-Gopalappa

File: app/controllers/claiimInviteCode.js (hunk 1 above)
Comment (srinizodiac) [reply app/controllers/claiimInviteCode.js]: fixed it

File: app/controllers/claiimInviteCode.js (hunk 3 above)
Comment (srinizodiac) [reply app/controllers/claiimInviteCode.js]: moved the code to service

File: app/controllers/claiimInviteCode.js (hunk 4 above)
Comment (srinizodiac) [reply app/controllers/claiimInviteCode.js]: fixed the code

File: app/controllers/claiimInviteCode.js (hunk 2 above)
Comment (srinizodiac) [reply app/controllers/claiimInviteCode.js]: created the type script files and moved the code. please review the code

Comment (srinizodiac) [reply app/controllers/claiimInviteCode.js]: can you review the typescript files created.

File: app/controllers/claiimInviteCode.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    const requestBody = JSON.parse(event.body);
+    var deviceId = context.deviceId;
+    var inviteCode = requestBody.inviteCode; 
+    var rubyAuthToken = event.requestContext.authorizer.claims['custom:ruby_auth_token'];
+    var rubyUserId =  event.requestContext.authorizer.claims['custom:ruby_user_id'];
+    var systemName =  requestBody.name;
+    var deviceType = requestBody.deviceType; 
+    
+    const claimInviteCodeService = new ClaimInviteCodeService();
+    var dynamoData: any;
```
Comment (bkoch-onica) [inline app/controllers/claiimInviteCode.ts]: Can we avoid the use of the 'any' type?

File: services/claimInviteCodeService.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
+import * as HttpStatus from "http-status-codes";  
+import * as config from "../config"
+import * as AWS from "aws-sdk"; 
+import * as requestPromise from "request-promise";
+
+
+export class ClaimInviteCodeService {
+
+ /**
```
Comment (chronicalley) [inline services/claimInviteCodeService.ts]: Add a constructor in this class that creates the dynamoDB object that's used by the other methods that write to Dynamo. Then you can inject a mock version of Dynamo DB when the class is created for the purposes of writing unit tests.

File: services/claimInviteCodeService.ts [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+ * 
+ * @param {string} rubyUserId the Ruby user id
+ * @param {string} rubyAuthToken the Ruby Authorization token
+ * @param {string} deviceId sent in the path of the request
+ * @param {string} deviceType sent in the body of the request type of device
+ * @param {string} systemName sent in the body of the request named by the user, not sure if used
+ * @return the result of the Ruby call
+ */
+
+public async addDeviceToLegacy(deviceId: string,deviceType: string,rubyUserId: any,rubyAuthToken: any,systemName: any) {
```
Comment (chronicalley) [inline services/claimInviteCodeService.ts]: Move this particular method (that does not write to Dynamo or do anything with Invite codes) to a separate class... Maybe something called `LegacyAPIClient` or similar.

File: app/controllers/inviteCodeGenerate.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+ * @param {String} inviteCode  the invite code
+ * @param {Integer} userId userID
+ * @param (DateTime) expiryDate expiryDate
+ * @returns params
+ */
+function recordParams(deviceId: any,inviteCode: any,userId: any,expiryDate: any) {
+
+    const record = {
+        deviceId: deviceId,
+        invite_code: inviteCode,
```
Comment (chronicalley) [inline app/controllers/inviteCodeGenerate.ts]: Let's settle on either camelCase or under_scores.. Don't need to mix them both.

File: app/controllers/inviteCodeGenerate.ts [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+ * ----------------------------
+ * This method creates the data in DynamoDB Table - invite_code
+ *
+ * @param {string} deviceId the device id
+ * @param {string} inviteCode the inviteCode
+ * @param {integer} userId the userId
+ * @param {datetime} expiryDate the expiryDate  
+ *  
+ */
+ function createDynamoDbData(deviceId: any,inviteCode: any,userId: any,expiryDate: any) {
```
[the remark this answers is not in the document]
Comment (chronicalley) [reply app/controllers/inviteCodeGenerate.ts]: Also this should be in the invite Code service class.

File: app/controllers/claiimInviteCode.ts (hunk 7 above)
Comment (srinizodiac) [reply app/controllers/claiimInviteCode.ts]: fixed as per comments

File: app/controllers/claiimInviteCode.ts [hunk 12]
```diff
@@ -0,0 +1,49 @@
+
+import * as config from "../../config"; 
+import * as AWS from "aws-sdk";  
+import {ClaimInviteCodeService} from "../../services/ClaimInviteCodeService";
+
+module.exports.handler = (event: any, context: any, callback: any) => {
+  try {
+
+    const requestBody = JSON.parse(event.body);
+    var deviceId = context.deviceId;
```
[the remark this answers is not in the document]
Comment (srinizodiac) [reply app/controllers/claiimInviteCode.ts]: fixed  as per comments

File: app/controllers/inviteCodeGenerate.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (srinizodiac) [reply app/controllers/inviteCodeGenerate.ts]: fixed as per comments

**Pull request merged**